tree-sitter/src/TreeSitter/Unmarshal/Examples.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE DeriveAnyClass, DeriveGeneric, DuplicateRecordFields, TypeOperators #-}
+module TreeSitter.Unmarshal.Examples () where
```

Comment:
Adding these examples to the library means that we get instantaneous feedback when changes to `Unmarshal` affect which instances we can derive.